z_infra_provisioning/cloud_infra_center/ocp_upi/roles/configure-bootstrap/tasks/main.yaml

```diff
@@ -76,4 +75,33 @@
     volume_size: "{{ bootstrap_flavor_size.stdout_lines[0]}}"
     terminate_volume: True
   when:
-  - disk_type == "scsi"
\ No newline at end of file
+  - disk_type == "scsi"
+  - use_storage_provider == false
```

Comment:
need @dongyanyang to help on this 
if I understand above correctly, disk_type = SCSI means it's BFV
use_stroage_provider = true means we will use storage provider anyway? 

so use_stroage_provider = true  must be BFV ?

Reply:
yes, I also have same confusion.

disk_type = SCSI already means BFV,  why would we need to add a new option use_storage_provider?

Reply:
We already supported the DASD and SCS two types in our OCP UPI earlier, for the SCSI disk_type, we just create the bootable VM and use the default volume template. 
For this new requirement of BFV, we changed to create volume firstly, and then launch instance to use this volume, enen though it is very similar.

Reply:
use_storage_provider is confusing, I think.
I understand that you want to differentiate between the two ways of BFV:
1. A single deploy from image. (which does not support specify volume type)
2. two-step deploy:  create bootable volume and then deploy. (can specify volume_type)

how about: 
when volume_type_id is specified by user, use way 2 to deploy.
Otherwise, goes to way 1.

Reply:
Great! I agree.

Reply:
thanks for your comments, it's already merged.